assets/app/scripts/services/auth.js

```diff
@@ -0,0 +1,226 @@
+  
+    return {
+      withUser: function() {
+        var user = userStore.getUser();
+        if (user) {
+          if (debug) { console.debug('AuthService.withUser()', user); }
+          var deferred = $q.defer();
+          deferred.resolve(user);
+          return deferred.promise;
+        } else {
+          if (debug) { console.debug('AuthService.withUser(), calling startLogin()'); }
+          return this.startLogin();
+        }
+      },
+  
+      setUser: function(user, token) {
+        if (debug) { console.debug('AuthService.setUser()', user, token); }
+        var oldUser = userStore.getUser();
+        userStore.setUser(user);
+        userStore.setToken(token);
+
+        var oldName = oldUser && oldUser.metadata && oldUser.metadata.name;
+        var newName = user    && user.metadata    && user.metadata.name;
+        if (oldName != newName) {
+          if (debug) { console.debug('AuthService.setUser(), user changed', oldUser, user); }
+          _userChangedCallbacks.fire(user);
+        }
+      },
+  
+      requestRequiresAuth: function(config) {
+        // TODO: replace with real implementation
+        var requiresAuth = config.url.toString().indexOf("api/") > 0;
+        if (debug && requiresAuth) { console.log('AuthService.requestRequiresAuth()', config.url.toString()); }
+        return requiresAuth;
+      },
+      addAuthToRequest: function(config) {
+        // Use the given token, if provided
+        var token = "";
+        if (config && config.auth && config.auth.token) {
+          token = config.auth.token;
+          if (debug) { console.debug('AuthService.addAuthToRequest(), using token from request config', token); }
+        } else {
+          token = userStore.getToken();
+          if (debug) { console.debug('AuthService.addAuthToRequest(), using token from user store', token); }
+        }
+        if (!token) {
+          if (debug) { console.debug('AuthService.addAuthToRequest(), no token available'); }
+          return false;
+        }
+  
+        // Handle web socket requests with a parameter
+        if (config.method == 'WATCH') {
+          config.url = URI(config.url).addQuery({access_token: token}).toString();
+          if (debug) { console.debug('AuthService.addAuthToRequest(), added token param', config.url); }
+        } else {
+          config.headers["Authorization"] = "Bearer " + token;
+          if (debug) { console.debug('AuthService.addAuthToRequest(), added token header', config.headers["Authorization"]); }
+        }
+        return true;
+      },
+    
+      startLogin: function() {
+        if (_loginPromise) {
+          console.log("Login already in progress")
+          return _loginPromise;
+        }
+        var self = this;
+        _loginPromise = loginService.login().then(function(user, token) {
+          self.setUser(user, token);
+          _loginCallbacks.fire(user);
+        }).catch(function(err) {
+          console.error(err);
```

Comment:
Service probably shouldn't redirect to an error page. If a controller was waiting on the loginPromise, it could handle an error by redirecting to the error page. Moot for now, since the callbacks aren't being used.